src/main/java/io/vertx/kafka/admin/ClusterDescription.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2019 Red Hat Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.vertx.kafka.admin;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+import io.vertx.kafka.client.common.Node;
+
+import java.util.Collection;
+
+/**
+ * A detailed description of the cluster
+ */
+@DataObject(generateConverter = true)
+public class ClusterDescription {
+  private String clusterId;
+  private Node controller;
+  private Collection<Node> nodes;
```

Comment:
why this class is missing the `Set<AclOperation>` from the native one?
https://kafka.apache.org/24/javadoc/org/apache/kafka/clients/admin/DescribeClusterResult.html

Reply:
I didn't include AclOperation in ClusterDescription as I didn't find the corresponding class in vertx kafka-client and I didn't know if I should introduce it, as it was also not used in vertx's TopicDescription. Please let me know how we should proceed. Shouldn't we create another PR specifically to include AclOperation everywhere where applicable?

Reply:
`AclOperation` is an enumeration. As for the `type` field in the `ConfigResource` class, you can use the native one. So, as we use `org.apache.kafka.common.config.ConfigResource.Type`, you could use `org.apache.kafka.common.acl.AclOperation` directly.

Reply:
Hi @ppatierno,
To retrieve the AclOperations from Kafka's adminClient, it seems that we need to call its describeCluster with a specific DescribeClusterOptions with its includeincludeAuthorizedOperations being set.
https://kafka.apache.org/24/javadoc/org/apache/kafka/clients/admin/DescribeClusterOptions.html
In this case, we get the AclOperations returned for the current test setup as [IDEMPOTENT_WRITE, CREATE, DESCRIBE_CONFIGS, DESCRIBE, CLUSTER_ACTION, ALTER_CONFIGS, ALTER]
However, without this include option being set to true, the AclOperations is empty. That means, to add Set<AclOperation> in ClusterDescription and make use of it, we either need to expose also describeCluster(DescribeClusterOperations) in vertx KafkaAdminClient or to set includeincludeAuthorizedOperations to true by default so that describeCluster() will fill the AclOperations.

By the way, vertx's kafka-client's other describe methods currently do not expose the corresponding operations with the DescribeOptions (e.g., describeTopics(Collection<String> topicNames) is exposed but not describeTopics(Collection<String> topicNames, DescribeTopicsOptions option)).

So, I suppose we have the following four possibilities:

1. leave AclOperation out for now in this PR and include it later
2. add Set<AclOperation> in ClusterDescription and the retrieval code but practically leaves it unfilled as includeincludeAuthorizedOperations being set to false.
3. add Set<AclOperation> in ClusterDescription and the retieval code and fills it with includeincludeAuthorizedOperations being set to true.
4. add it and expose descriptCluster(DescribeClusterOperations) so that the user can decide whether to retrieve it or not.

Do you think we should go for 4 with this PR or stay with 1 or 2 and have another PR to go for 4 for other describe operations? I would prefer the latter and I would be glad to work on it too. Please let me know how you think. Thanks.

regards, aki

Reply:
Because as you said, we don't expose options parameter for other "describeXXX" methods I would do 3 for now, retrieving the AclOperation(s) by default. We can add the methods with options in the future as an improvement.

Reply:
@ppatierno Actually, option 3 is not as consistent as the other 3 options and that is why I suggested that we would rather go for either 1 or 2. This option 3 would lead to a change of behavior when we later add the methods with the options argument and want to have the behavior of this no-args method to be aligned with that of the original method. In contrast, staying with options 1 or 2 would not interfere with the behavior of the original methods now and for the future. How do you think?

Reply:
Good point. I think that option 2 is not so well understood by the user which will get an always empty list. I would go for 1 at this point.

Reply:
sounds good. That mens that this PR is ready?